pkg/prowgen/prowgen.go

```diff
@@ -333,18 +333,39 @@ func generatePodSpecMultiStage(info *ProwgenInfo, test *cioperatorapi.TestStepCo
 		secrets = append(secrets, &api.Secret{Name: api.HiveControlPlaneKubeconfigSecret})
 	}
 	podSpec := generateCiOperatorPodSpec(info, secrets, []string{test.As}, additionalArgs...)
-	if profile == "" {
-		return podSpec
+
+	if profile != "" {
+		podSpec.Volumes = append(podSpec.Volumes, generateClusterProfileVolume(profile, profile.ClusterType()))
+		clusterProfilePath := fmt.Sprintf("/usr/local/%s-cluster-profile", test.As)
+		container := &podSpec.Containers[0]
+		container.Args = append(container.Args, fmt.Sprintf("--secret-dir=%s", clusterProfilePath))
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{Name: "cluster-profile", MountPath: clusterProfilePath})
 	}
-	podSpec.Volumes = append(podSpec.Volumes, generateClusterProfileVolume(profile, profile.ClusterType()))
-	clusterProfilePath := fmt.Sprintf("/usr/local/%s-cluster-profile", test.As)
-	container := &podSpec.Containers[0]
-	container.Args = append(container.Args, fmt.Sprintf("--secret-dir=%s", clusterProfilePath))
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{Name: "cluster-profile", MountPath: clusterProfilePath})
-	addLeaseClient(podSpec)
+
+	if testContainsLease(test) || profile != "" {
```

Comment:
Reverse the predicates since the second is much less expensive to compute and can likely be combined with the previous `if`.

Reply:
Yeah, good call.